coverage/PENDING_REVIEW/intel/0439-COVERAGE-nir-opt-constant-folding-238-340-nir-range-analysis-866/cov-min-nested-loop-same-value-for-variables.amber

```diff
@@ -0,0 +1,434 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex texgen_vert PASSTHROUGH
+
+SHADER fragment texgen_frag GLSL
+#version 430
+precision highp float;
+
+layout(location = 0) out vec4 _GLF_color;
+
+void main()
+{
+ _GLF_color = vec4(
+ gl_FragCoord.x * (1.0 / 256.0),
+ (int(gl_FragCoord.x) ^ int(gl_FragCoord.y)) * (1.0 / 256.0),
+ gl_FragCoord.y * (1.0 / 256.0),
+ 1.0);
+}
+END
+BUFFER default_texture FORMAT B8G8R8A8_UNORM
+
+PIPELINE graphics texgen_pipeline
+  ATTACH texgen_vert
+  ATTACH texgen_frag
+  FRAMEBUFFER_SIZE 256 256
+  BIND BUFFER default_texture AS color LOCATION 0
+END
+
+CLEAR_COLOR texgen_pipeline 0 0 0 255
+CLEAR texgen_pipeline
+RUN texgen_pipeline DRAW_RECT POS 0 0  SIZE 256 256
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# #define _int_1 _GLF_uniform_int_values[0]
+# #define _int_0 _GLF_uniform_int_values[1]
+# #define _float_1_0 _GLF_uniform_float_values[0]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_float_values: 1.0
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     float _GLF_uniform_float_values[1];
+# };
+# // Contents of _GLF_uniform_int_values: [1, 0]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     int _GLF_uniform_int_values[2];
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+# layout(set = 0, binding = 2) uniform sampler2D tex;
+#
+# void main()
+# {
+#     const float arr[9] = float[9](0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+#     vec2 coord = gl_FragCoord.xy;
+#     vec4 res = vec4(_int_1, _int_0, _int_0, _int_1);
+#
+#     for (int i = 0; i < 3; i++)
+#     {
+#         for (int j = 0; j < 3; j++)
+#         {
+#             int a = clamp(i + j, 0, 9);
+#             float f0 = clamp(arr[a], arr[a], arr[a]);
+#             float f1 = clamp(arr[a], arr[a], arr[a]);
+#             vec4 v0 = vec4(f0);
+#             vec4 v1 = texture(tex, vec2(i)) * f1;
+#             vec4 v2 = texture(tex, coord) * arr[a];
+#             vec4 v3 = texture(tex, coord) * arr[a];
+#             vec4 v4 = texture(tex, coord + vec2(i - _int_1, j - _int_1)) * arr[a];
```

Comment:
When `i` and `j` are 0 this will index into the texture out of bounds, but that's OK I suppose. I don't know too much about textures and samplers. Can you check that this doesn't lead to undefined behaviour that would make the test invalid?

Reply:
When sampling textures out of the UV space of 0..1 it will behave based on the sampler wrapping mode. It can either repeat the texture, return a border color, or the color value at the edge of the texture. In any case it's perfectly fine to index with negative values.